Sources/StreamChatUI/CommonViews/Attachments/DefaultAttachmentPreviewProvider.swift

```diff
@@ -0,0 +1,45 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// Default provider that is used when AttachmentPreviewProvider is not implemented for custom attachment payload. This
+/// provider always returns a new instance of `AttachmentPlaceholderView`.
+public struct DefaultAttachmentPreviewProvider: AttachmentPreviewProvider {
+    public func previewView<ExtraData: ExtraDataTypes>(components: _Components<ExtraData>) -> UIView {
+        components.attachmentPreviewViewPlaceholder.init()
+    }
+
+    public static var preferredAxis: NSLayoutConstraint.Axis { .horizontal }
+}
+
+/// Default attachment view to be used as a placeholder when attachment preview is not implemented for custom attachments.
+open class AttachmentPlaceholderView: _View, AppearanceProvider {
+    open private(set) lazy var borderLayer: CAShapeLayer = .init()
+
+    override open func layoutSubviews() {
+        super.layoutSubviews()
+
+        borderLayer.frame = bounds
+        borderLayer.path = UIBezierPath(roundedRect: bounds, cornerRadius: 16).cgPath
```

Comment:
This is probably overthinking 😅 but if we want to make our API even more convenient, we could extract some cornerRadius types and create some extension on float that we would type: cornerRadius: .halfOfheight(of: view) or cornerRadius: .medium Currently if you take a look into the project there is really lot of different corner-radiuses and it feels it's not unified across views... Maybe we can discuss this with designers later to make this nicer and for our customers not to override anything when wanting to unify those..

 But this is not important right now

Reply:
Yeah, good point 👍 Feel free to create a JIRA ticket for it